daemon/image_service.go

```diff
@@ -0,0 +1,70 @@
+package daemon
+
+import (
+	"context"
+	"io"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/backend"
+	"github.com/docker/docker/api/types/filters"
+	imagetype "github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/api/types/registry"
+	"github.com/docker/docker/builder"
+	"github.com/docker/docker/container"
+	"github.com/docker/docker/daemon/images"
+	"github.com/docker/docker/image"
+	"github.com/docker/docker/layer"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
```

Comment:
Perhaps we should add a big TODO here, with "this interface is considered temporary"